resources/Microsoft.Windows.Setting.Time/Microsoft.Windows.Setting.Time.psm1

```diff
@@ -0,0 +1,376 @@
+if ([string]::IsNullOrEmpty($env:TestRegistryPath)) {
+    $global:tzAutoUpdatePath = 'HKLM:\SYSTEM\CurrentControlSet\Services\W32Time\Parameters'
+    $global:SysTrayPath = 'HKCU:\Software\Microsoft\Windows\CurrentVersion\Explorer\Advanced'
+    $global:AdditionalClockPath = 'HKCU:\Control Panel\TimeDate'
+} else {
+    $global:tzAutoUpdatePath = $global:SysTrayPath = $global:AdditionalClockPath = $env:TestRegistryPath
+}
+
+#region Functions
+function TryGetRegistryValue {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$Key,
+
+        [Parameter(Mandatory = $true)]
+        [string]$Property
+    )
+
+    if (Test-Path -Path $Key) {
+        try {
+            return (Get-ItemProperty -Path $Key -Name $Property -ErrorAction SilentlyContinue | Select-Object -ExpandProperty $Property)
+        } catch {
+            Write-Verbose "Property `"$($Property)`" could not be found."
+        }
+    } else {
+        Write-Verbose 'Registry key does not exist.'
+    }
+}
+#endRegion Functions
+
+#region Enums
+enum TimeZoneTable {
```

Comment:
This enum seems like overkill. I would just let the time zone id be used directly as a string parameter instead of creating this enum.